### PR TITLE
added loss parameter

### PR DESCRIPTION
**Pull request by spookieoli**
Added minibatches and interchangable loss functions to neural net trainer, you can toggle training with:

type Classifier struct {
	ApiKey         string  `json:"api_key"`
	CollectionName string  `json:"collection_name"`
	ClassifierName string  `json:"classifier_name"`
	Degree         int     `json:"degree"`
	C              float64 `json:"c"`
	Epochs         int     `json:"epochs"`
	Type           string  `json:"type"`
	Loss           string  `json:"loss"`
	Batchsize      int     `json:"batchsize"`
}

For neural net you would use:

type Classifier struct {
	ApiKey         string  `json:"api_key"`
	CollectionName string  `json:"collection_name"`
	ClassifierName string  `json:"classifier_name"`
	Epochs         int     `json:"epochs"` #  any number you wish
	Type           string  `json:"type"` # "NN"
	Loss           string  `json:"loss"` # e.g "mae" or "mse"
	Batchsize      int     `json:"batchsize"`# any number you wish
}

### CODE REVIEW

**Review by RamBITS-AI — APPROVED** [review]
Just found one issue, I guess. Please check it out.

File: NN/NeuralNet.go [hunk 1]
```diff
@@ -99,15 +135,26 @@ func (n *Network) Train(trainingData [][]float64, targets [][]float64, epochs in
...
+			}
+			batchData := trainingData[i:end]
+			batchTargets := targets[i:end]
+
+			// Train on batch
+			totalLoss := 0.0
+			for i, input := range batchData {
+				output := n.Forward(input)
+				n.Backpropagate(input, batchTargets[i], lr)
+				totalLoss += n.MSE(output, batchTargets[i])
```
Comment (RamBITS-AI) [inline NN/NeuralNet.go:152]:
```suggestion
				totalLoss += n.Loss(output, batchTargets[i])
```

Comment (spookieoli) [reply NN/NeuralNet.go]: Thanks a lot - fixed!

**Comment by RamBITS-AI** [comment]
Sorry for this delayed comment. Please check this as well.
